cmd/horcrux/cmd/config.go

```diff
@@ -218,13 +484,69 @@ type CosignerPeer struct {
 	P2PAddr string `json:"p2p-addr" yaml:"p2p-addr"`
 }
 
+func readKeyShare() (*signer.CosignerKey, error) {
+	// Read ID from local key share
+	var home string // In root.go we end up with our
+	if homeDir != "" {
+		home = homeDir
+	} else {
+		home, _ = homedir.Dir()
+		home = path.Join(home, ".horcrux")
+	}
+
+	if _, err := os.Stat(homeDir); !os.IsNotExist(err) {
+		return nil, fmt.Errorf("%s is not empty, check for existing configuration and clear path before trying again", homeDir)
+	}
+
+	file, err := os.Open(path.Join(home, "share.json"))
+	if err != nil {
+		return nil, err
+	}
+	defer file.Close()
+
+	bz, err := ioutil.ReadAll(file)
+	if err != nil {
+		return nil, err
+	}
+	var key signer.CosignerKey
+	if err := json.Unmarshal(bz, &key); err != nil {
+		return nil, err
+	}
+	return &key, nil
+}
+
 func validateCosignerPeers(peers []CosignerPeer) error {
-	// TODO: check IDs to ensure none are duplicated and
-	// that the key share which is configured for the local node
+	// Check IDs to make sure none are duplicated
+	if dupl := duplicatePeers(peers); len(dupl) != 0 {
+		return fmt.Errorf("found duplicate share IDs in args: %v", dupl)
+	}
+
+	// Check that the key share which is configured for the local node
 	// is the last remaining share
```

Comment:
My initial approach was to just go through the peers in the config and check for the missing ID, which assumes that IDs are increments of 1. Since I don't know if that's safe to assume, I just read the share ID from the `share.json`.

However, that means that the `share.json` needs to be present before I can use `horcrux config init`, which requires creating the config directory manually, so that's obviously pointless.

So my question is, is it safe to assume that the share IDs are always increments of 1, starting from ID 1?

Reply:
Yes it is safe to assume that share id's are 1 indexed and incrementing up from 1